quicklens/src/main/scala-3/com/softwaremill/quicklens/QuicklensMacros.scala

```diff
@@ -167,49 +163,46 @@ object QuicklensMacros {
       def widenAll: TypeRepr =
         tpe.widen.dealias.poorMansLUB
 
-      def matchingTypeSymbol: Symbol = {
-        def recurse(tpe: TypeRepr): Symbol = {
-          tpe.widenAll match {
-            case AndType(l, r) =>
-              val lSym = recurse(l)
-              if lSym != Symbol.noSymbol then lSym else recurse(r)
-            case tpe if isProduct(tpe.typeSymbol) || isSum(tpe.typeSymbol) =>
-              tpe.typeSymbol
-            case tpe if isProductLike(tpe.typeSymbol) =>
-              tpe.typeSymbol
-            case _ =>
-              Symbol.noSymbol
-          }
-        }
-        val rSym = recurse(tpe)
-        if rSym != Symbol.noSymbol then rSym else tpe.typeSymbol // if everything else fails, try the original type, maybe it will have all we need
+      def matchingTypeSymbol: Symbol = tpe.widenAll match {
+        case AndType(l, r) =>
+          val lSym = l.matchingTypeSymbol
+          if lSym != Symbol.noSymbol then lSym else r.matchingTypeSymbol
+        case tpe if isProduct(tpe.typeSymbol) || isSum(tpe.typeSymbol) || isProductLike(tpe.typeSymbol) =>
+          tpe.typeSymbol
+        case _ =>
```

Comment:
Here you have removed the fallback to the original type. The idea behind it was: we do not need a case class, we just need anything which has copy and member access. If there is a type we do not understand, maybe we can just go straight forward, the worst which can happen is it will fail later. It was partially helpful with some opaque types I was trying to implement, but I do remember a single particular example and certainly it is not necessary for the code in tests.

Reply:
Yes, but the implementation will not handle such a type, these checks are exactly the same as when generating the `copy` calls.